pkg/api/mock/matcher_object.go

```diff
@@ -0,0 +1,50 @@
+package mock
+
+import (
+	"fmt"
+	"reflect"
+
+	gomegaTypes "github.com/onsi/gomega/types"
+	"go.anx.io/go-anxcloud/pkg/api/types"
+	"go.anx.io/go-anxcloud/pkg/utils/object/compare"
+)
+
+// ObjectMatcher matches mockAPIObject with provided Object
+// It compares Objects by matchFields
+func ObjectMatcher(obj types.Object, matchFields ...string) gomegaTypes.GomegaMatcher {
+	return &objectMatcher{
+		obj:         obj,
+		matchFields: matchFields,
+	}
+}
+
+type objectMatcher struct {
+	obj         types.Object
+	matchFields []string
+}
+
+func (m *objectMatcher) Match(actual interface{}) (bool, error) {
+	obj, ok := actual.(*mockAPIObject)
+	if !ok {
+		return false, fmt.Errorf("ObjectMatcher matcher expects an *mockAPIObject")
+	}
+
+	if reflect.TypeOf(m.obj) != reflect.TypeOf(obj.Object) {
+		return false, nil
+	}
+
+	diff, err := compare.Compare(obj.Object, m.obj, m.matchFields...)
+	if err != nil {
+		return false, err
+	}
+
+	return len(diff) == 0, nil
+}
+
+func (m *objectMatcher) FailureMessage(actual interface{}) string {
+	return "Matched Object is not present"
```

Comment:
first: this is a matcher for comparing a single ACTUAL Object with a single EXPECTED Object, so the message is more like "ACTUAL Object does not match EXPECTED Object"

second: having the difference printed would be very helpful, implementing this with an existing primitive matcher could help again

Reply:
hm re $second: using an existing primitive matcher is probably hard, but you can extract that info from `diff` returned by `compare.Compare`

that might mean executing `compare.Compare()` multiple times - for checking and for message, maybe you find a clever way of preventing that though